sql/src/main/java/io/crate/operation/projectors/DistributingDownstreamFactory.java

```diff
@@ -112,4 +110,22 @@ public BatchConsumer create(NodeOperation nodeOperation,
             pageSize
         );
     }
+
+    /**
+     * @return bucketIdx (= idx of localNode in nodeIds)
+     *
+     * <p>
+     * This needs to be deterministic across all nodes involved in a query execution.
+     *
+     * Downstreams assume that each upstream uses a *different* bucketIdx,
```

Comment:
What is used as an offset to ensure that each upstream node can generate a unique id?

Reply:
No offset required,  nodeIds.indexOf(localNodeId) is unique. (Each node that is contained within the nodeIds of a NodeOperation, will receive that *same* NodeOperation) - Should I clarify that somehow in the comments here?

Reply:
Never mind, saw that the local node id is used.